amlb/runners/docker.py

```diff
@@ -31,17 +30,23 @@ def __init__(self, framework_name, benchmark_name, constraint_name):
         """
         super().__init__(framework_name, benchmark_name, constraint_name)
         self._custom_image_name = rconfig().docker.image
+        self.user = os.getlogin()
+        self.userid = os.getuid()
+        self.usergid = os.getgid()
         self.minimize_instances = rconfig().docker.minimize_instances
-        self.container_name = 'docker'
+        self.container_name = "docker"
```

Comment:
> @alanwilter, to be clear, this fixes issues introduced with #464, correct?
> 
> Could you please detail a bit more the kind of issues that were introduced by this? For us to understand the changes, for other users that may face the same problems… thanks.

Yes, I found out that when running docker and docker line `RUN $PY {script} {framework} -s only` was called, some frameworks need to install ubuntu packages etc. and this was failing.

Now all runs as the local `user` issuing the docker cmd and will save all for that `user`, and this `user` is enabled to run `sudo` inside the docker, without password.

The other issue addressed was that, again, when docker running `RUN $PY {script} {framework} -s only`, for some frameworks, directive `ENV DEBIAN_FRONTEND noninteractive` was not working to avoid `tzdata` to enter in the interactive mode. So, I found better to install it at the first opportunity. You can rearrange it as you wish.

I found this issue when debugging `autoxgboost`, which installs 

```bash
  SUDO apt-get install -y r-base r-base-dev
  SUDO apt-get install -y libgdal-dev libproj-dev
  SUDO apt-get install -y libssl-dev libcurl4-openssl-dev
  SUDO apt-get install -y libcairo2-dev libudunits2-dev
```
and one of these pkgs has `tzdata` dependency.